src/main.cpp

```diff
@@ -115,748 +160,765 @@ int main ( int argc, char** argv )
-                               "--showallservers" ) )
-        {
-            bShowComplRegConnList = true;
-            qInfo() << "- show all registered servers in server list";
-            CommandLineOptions << "--showallservers";
-            ClientOnlyOptions << "--showallservers";
-            continue;
-        }
+            // Mute stream on startup ----------------------------------------------
+            if ( GetFlagArgument ( argv, i, "-M", "--mutestream" ) )
+            {
+                bMuteStream = true;
+                qInfo() << "- mute stream activated";
+                CommandLineOptions << "--mutestream";
+                ClientOnlyOptions << "--mutestream";
+                continue;
+            }
 
-        // Show analyzer console -----------------------------------------------
-        // Undocumented debugging command line argument: Show the analyzer
-        // console to debug network buffer properties.
-        if ( GetFlagArgument ( argv,
-                               i,
-                               "--showanalyzerconsole", // no short form
-                               "--showanalyzerconsole" ) )
-        {
-            bShowAnalyzerConsole = true;
-            qInfo() << "- show analyzer console";
-            CommandLineOptions << "--showanalyzerconsole";
-            ClientOnlyOptions << "--showanalyzerconsole";
-            continue;
-        }
+            // For headless client mute my own signal in personal mix --------------
+            if ( GetFlagArgument ( argv,
+                                   i,
+                                   "--mutemyown", // no short form
+                                   "--mutemyown" ) )
+            {
+                bMuteMeInPersonalMix = true;
+                qInfo() << "- mute me in my personal mix";
+                CommandLineOptions << "--mutemyown";
+                ClientOnlyOptions << "--mutemyown";
+                continue;
+            }
 
-        // Unknown option ------------------------------------------------------
-        qCritical() << qUtf8Printable ( QString ( "%1: Unknown option '%2' -- use '--help' for help" ).arg ( argv[0] ).arg ( argv[i] ) );
+            // Client Name ---------------------------------------------------------
+            if ( GetStringArgument ( argc,
+                                     argv,
+                                     i,
+                                     "--clientname", // no short form
+                                     "--clientname",
+                                     strArgument ) )
+            {
+                strClientName = strArgument;
+                qInfo() << qUtf8Printable ( QString ( "- client name: %1" ).arg ( strClientName ) );
+                CommandLineOptions << "--clientname";
+                ClientOnlyOptions << "--clientname";
+                continue;
+            }
 
-// clicking on the Mac application bundle, the actual application
```

Comment:
Explain in the commit message why removing this is safe (presumably it wasn't reached originally because of the `qCritical` - but I'd check the Git history to see why it was added).

Reply:
Yes, this was a strange exception....
Since 1. I tested it on macOS and didn't see any strange behavior, and 2. the same exit ( 1 ) is used in multiple places, so if it really was a problem it should have been implemented in multiple places...

But yeah, I should have mentioned it...